### ISSUE TITLE
Incorrect highlighting of URLs

### ISSUE THREAD

**Issue by vhyrro** [nvim-neorg/neorg#53]
There's a bug where URLs get concealed absolutely awfully when in a TODO item :P

![error](https://user-images.githubusercontent.com/76052559/127771167-f4b4acc4-3b50-45de-a4cb-5c6e35d86b28.png)

This is because the regex used for concealing skips over this character as it doesn't have a parenthesis after itself rather than stopping parsing at that point

**Comment by vhyrro** [comment]
I tried to address this issue, but there's still a problem that might be a physical limitation of regex. Trying to escape a ] character (via `\]`) in the first section of the link causes the entire link to stop being highlighted entirely. I've spent 2 hours on this single issue alone so I'm gonna leave it be.

If there's anyone who believes they can do some regex-fu then please check out [what I tried doing](https://github.com/vhyrro/neorg/blob/unstable/lua/neorg/modules/core/norg/concealer/module.lua#L400) and see whether you can come up with any extra solutions :)

**Comment by mrossinek** [comment]
Using the latest `unstable` branch I now get the following concealing + highlighting for the following text:
![screenshot_1627939015](https://user-images.githubusercontent.com/21973473/127925061-394a973d-9b8f-4a5f-b572-cef66a3529a0.png)
```
[link](https://archwiki.org)
- [*] This is a [test link](https://archwiki.org)
```
So the URL inside of the TODO works as expected but the plain one does not.
However, while typing this I did notice that a single space before the link seems to fix the pure URL concealing+highlighting..

**Comment by vhyrro** [comment]
That's something i observed too, it's got to do with the lookback the regex does. I'll see whether there's anything i can do about it tomorrow :)

**Comment by vhyrro** [comment]
Just an FYI since the parser is being rewritten to detect bold, italic, links etc. concealing and highlighting should soon become trivial - no more erroneous highlights 🤞

**Comment by mrossinek** [comment]
This was fixed in #85 but somehow not closed by the commit keyword.

**Comment by danymat** [comment]
> This was fixed in #85 but somehow not closed by the commit keyword.

Maybe the correct syntax was: `fix #53`

**Comment by mrossinek** [comment]
That's what the commit message reads: https://github.com/mrossinek/neorg/commit/645841cc7ff91e7882f268dacbd633a91bbf702e.patch

But Github now renders it slightly differently :shrug:
